core/runtime/src/main/java/io/quarkiverse/flow/providers/CredentialsProviderSecretManager.java

```diff
@@ -0,0 +1,153 @@
+package io.quarkiverse.flow.providers;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.TreeSet;
+import java.util.concurrent.ConcurrentHashMap;
+
+import jakarta.annotation.PostConstruct;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.enterprise.inject.Any;
+import jakarta.enterprise.inject.Instance;
+import jakarta.enterprise.inject.literal.NamedLiteral;
+import jakarta.inject.Inject;
+import jakarta.inject.Named;
+
+import io.quarkiverse.flow.config.FlowSecretsConfig;
+import io.quarkus.credentials.CredentialsProvider;
+import io.serverlessworkflow.impl.config.SecretManager;
+
+/**
+ * SecretManager bridge that sources Workflow Secrets from Quarkus CredentialsProvider beans.
+ * Selection rules:
+ * <ol>
+ * <li>per-secret override: quarkus.flow.secrets.credentials-provider-names.&gt;secret&lt;=@NamedBean</li>
+ * <li>global: quarkus.flow.secrets.credentials-provider-name=@NamedBean</li>
+ * <li>single provider on classpath → use it</li>
+ * <li>else fail with guidance</li>
+ * </ol>
+ */
+@ApplicationScoped
+public class CredentialsProviderSecretManager implements SecretManager {
+
+    private static final String ROOT = FlowSecretsConfig.ROOT_KEY;
+    private static final String GLOBAL_KEY = ROOT + ".credentials-provider-name";
+    private static final String PER_SECRET_KEY = ROOT + ".credentials-provider-names.<secret>";
+    private final Map<String, CredentialsProvider> providerBeanCache = new ConcurrentHashMap<>();
```

Comment:
This one does not need to be concurrent, since it is initialized at build time. Note that An inmutable map that is concurrently accesed does not need to be concurrent, so using HashMap is faster and safe